web/app/components/header/mobileHeader.tsx

```diff
@@ -0,0 +1,96 @@
+import { Grid, Text, Burger, Autocomplete, Flex } from "@mantine/core"
+import { IconGrillFork, IconSearch } from "@tabler/icons-react"
+import { useDisclosure } from "@mantine/hooks"
+import { useState, useEffect } from "react"
+import { Navigation } from "../navigation/navigation"
+import { Categories } from "../categoriesList/categories"
+
+export function MobileHeader() {
+  const [opened, { toggle }] = useDisclosure()
+  const [isVisible, setIsVisible] = useState(false)
+
+  useEffect(() => {
+    if (isVisible) {
+      document.body.style.overflow = "hidden" // Disable scrolling
+    } else {
+      document.body.style.overflow = "" // Re-enable scrolling
+    }
+
+    // Clean up on component unmount
+    return () => {
+      document.body.style.overflow = ""
+    }
+  }, [isVisible])
```

Comment:
I don't think I fully understand what this is doing? Is it to disable scrolling on the body? I think if we use the Mantine `Drawer` component it will handle that for us.